lib/realtime/nodes.ex

```diff
@@ -0,0 +1,130 @@
+defmodule Realtime.Nodes do
+  @moduledoc """
+  Handles common needs for :syn module operations
+  """
+  require Logger
+  alias Realtime.Tenants
+  alias Realtime.Api.Tenant
+
+  @doc """
+  Gets the node to launch the Postgres connection on for a tenant id.
+  """
+  @spec get_node_for_tenant_id(String.t()) :: node()
```

Comment:
This spec does not match the implementation fully. It may return `{:error, ...}` or even `nil` (which is a valid node typing wise but never in practice).